docs/releases/stable/index.md

```diff
@@ -0,0 +1,192 @@
+
+| base                     | python |
+| ---                      | ---    |
+| ubuntu@20.04 (**focal**) | 3.8    |
+| ubuntu@22.04 (**jammy**) | 3.10   |
+| ubuntu@24.04 (**noble**) | 3.12   |
+
+Choose the environment based build-on base in `charms/worker/k8s/charmcraft.yaml`
+
+In the following example, building on focal yields packages for py38.
+```yaml
+  - build-on:
+    - name: ubuntu
+      channel: "20.04"
+      architectures: [amd64]
+```
+
+So create a py38 environment, and freeze those libraries
+Then create a PR to merge into the release branch
+
+Task:
+```sh
+pyenv install 3.8
+pyenv virtualenv 3.8 k8s-operator
+pyenv activate k8s-operator
+git switch release_1.xx
+git checkout -b task/pip-pinning/release-1.xx
+pip install -r charms/worker/k8s/requirements.txt
+pip freeze > charms/worker/k8s/requirements.txt
+```
+
+### Build charms from the release branches
+
+Raising a PR, passing the integration tests, and merging into the release
+branch should publish the charm to the upstream `1.xx/beta` channel
+
+## Internal verification
+
+### Run **validate-k8s-release-upgrade** job
+
+**Job**: https://jenkins.canonical.com/k8s-ps5/job/validate-k8s-release-upgrade/
+
+This validates the deployment using charms from the `$prev/stable` channel,
+then performing an upgrade to `1.xx/beta`. The tests are parameterized to
+run on 
+* multiple series
+* multiple architectures
+* multiple clouds (aws/azure/gcp/vsphere)
+
+### Notify Solutions QA
+
+At the end of the first week and assuming all major blockers are resolved, the
+release is passed over to Solutions QA (SQA) for sign-off. This is done by
+[publishing a CI release](https://github.com/charmed-kubernetes/jenkins/releases/new)
+with a new `1.xx` tag and informing SQA of that tag. They will then have the
+remaining week to test and file bugs so engineering can work towards getting
+them resolved prior to GA.
+
+### CNCF Conformance
+
+**Job**: https://jenkins.canonical.com/k8s-ps5/job/conformance-cncf-k8s/
+
+Sync `canonical/k8s-conformance` main from upstream
+
+- https://github.com/canonical/k8s-conformance
+
+Confirm passing results, then create a PR against the upstream `k8s-conformance`
+repo.
+
+And opened an upstream PR:
+
+- https://github.com/cncf/k8s-conformance/pull/XXXXX
+
+> **Note**: CNCF requires a sign-off. After confirming results, issue a
+`git commit --amend --signoff` on the branch prior to submitting the PR.
+
+## Performing the release
+
+### Document release notes
+
+- Bugfixes
+- Enhancements
+- Known Limitations/Issues
+
+### Promote charms to stable
+**Job**: https://github.com/canonical/k8s-operator/actions/workflows/promote_charm.yaml
+
+Run the workflow from a branch, select `release_1.xx`, 
+* Choose `Charm` - `all`
+* Choose `Origin Channel`- `beta`
+* Choose `Destination Channel` - `stable`
+
+![promote charm options](promote-charm.png)
+
+### Send announcement
+
+Email announcement to k8s-crew with any relevant information.
+
```

Comment:
should we also add a section about how backports are handled in this release process?

Reply:
Cool, i'll add another document for this called `bug-fix`